scripts/assign_taxonomy_from_blast/ClassifyContigNR.py

```diff
@@ -126,193 +152,271 @@ def map_gids_binary(gids, mapping_file):
+            assign_file.write('\n')
+            support_file.write('\n')
+            assign_file.flush()
+            support_file.flush()
+
+
+def write_contig_assigns(output_dir, contigAssign, contigLengths):
+    with open(output_dir + "_contigs.csv", "w") as assign_file, open(output_dir+"_contigs.supports.csv", "w") as support_file:
         for contig in contigAssign.keys():
-            text_file.write('%s,%f'%(contig,contigLengths[contig]))
+            assign_file.write('%s,%f' % (contig, contigLengths[contig]))
             for depth in range(7):
-                (assign,p,dF) = contigAssign[contig][depth]
-                dFN = dF/contigLengths[contig]
-                text_file.write(',%s->%.3f->%.3f'%(assign,p,dFN))
-            text_file.write('\n')
-            text_file.flush()
-    
-if __name__ == "__main__":
-    main(sys.argv[1:])
+                (assign, p, dF) = contigAssign[contig][depth]
+                dFN = dF / contigLengths[contig]
+                assign_file.write(',%s'%assign)
+                support_file.write(',%.3f/%.3f' % (p, dFN))
+            assign_file.write('\n')
+            support_file.write('\n')
+            assign_file.flush()
+            support_file.flush()
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--blast_input_file",
+                        help="Blast input file with blast 6 matches to taxaid database *.b6")
+    parser.add_argument('-g', '--gid_taxaid_mapping_file',
+                        help="mapping from gid to taxaid gzipped")
+    parser.add_argument('-a', '--acc_taxaid_mapping_file',
+                        help="mapping from accession to taxaid gzipped")
+    parser.add_argument('-b', '--bed_file', required=True,
+                        help="bed file defining genes on contigs")
+    parser.add_argument('-l', '--lineage_file', required=True,
+                        help="text taxaid to lineage mapping")
+    parser.add_argument('-f', '--min_fraction', default=0.5, type=float,
+                        help="Minimum fraction of weights needed to assign to a particular level, default 0.5. 0.9 \
+                        would be more strict.")
+    parser.add_argument('-m', '--min_id', default=40.0, type=float,
+                        help="Minimum allowed percent identity to parse a hit")
+    parser.add_argument('-o', '--output_dir', type=str, default="output",
+                        help=("string specifying output directory and file stubs"))
+    parser.add_argument("-t", "--taxa_identities", nargs="*", default=[0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.95],
+                        help="Specify normalized weights for the ranks: superkingdom phylum class order\
+                         family genus species. Defaults to 0.4 0.5 0.6 0.7 0.8 0.9 0.95")
+
+    args = parser.parse_args()
+
+    default_taxa_identities = [0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.95]
+    taxa_identities = args.taxa_identities
+    # Add identities if not all specified
```

Comment:
I think it's better to demand a array of length 7 or none at all of the user. As it is now, the user might specify 1,1,1,1,0.3,02 and then it will be completed with 0.95. It can easily get weird.